matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java

```diff
@@ -2102,7 +2102,107 @@ public void sendEvent(final Event event, final ApiCallback<Void> callback) {
 
         final String prevEventId = event.eventId;
 
-        final ApiCallback<CreatedEvent> localCB = new ApiCallback<CreatedEvent>() {
+        final ApiCallback<CreatedEvent> localCB = getLocalCB(event, callback, prevEventId);
```

Comment:
I feel like the extraction here helps a bit with readability. I think I see why you did not extract it into a different class, I assume the security is the reason for that. But still, this extraction into a function helps.